src/app/store/accounts/blockchain/bitcoin/bitcoin-accounts.hooks.ts

```diff
@@ -0,0 +1,24 @@
+import { useMemo } from 'react';
+
+import { useCurrentAccountNativeSegwitSigner } from './native-segwit-account.hooks';
+
+export interface BtcAccount {
+  index: number;
+  address: string;
+}
+
+export function useBtcAccounts(): BtcAccount[] {
+  const signer = useCurrentAccountNativeSegwitSigner();
+
+  return useMemo(() => {
+    if (!signer) return [];
+    const addresses = [];
+    for (let i = 0; i < 5; i++) {
+      const address = signer(i).payment.address;
+      if (address) {
+        addresses.push({ index: i, address });
+      }
+    }
+    return addresses;
+  }, [signer]);
+}
```

Comment:
Didn't we find a way to not ever need to do this by just checking for account when needed, rather than creating array first?

Reply:
we need to check how many accounts there are in list for virtuoso
```
      <Virtuoso
        initialTopMostItemIndex={whenWallet({ ledger: 0, software: currentAccountIndex })}
        height="72px"
        style={{ paddingTop: '24px', height: '70vh' }}
        totalCount={stacksAccounts.length || bitcoinAccounts.length}
```